homeassistant/components/google_assistant/trait.py

```diff
@@ -362,23 +363,26 @@ class ColorSettingTrait(_Trait):
     commands = [COMMAND_COLOR_ABSOLUTE]
 
     @staticmethod
-    def supported(domain, features, device_class):
+    def supported(domain, features, device_class, attributes):
         """Test if state is supported."""
         if domain != light.DOMAIN:
             return False
 
-        return features & light.SUPPORT_COLOR_TEMP or features & light.SUPPORT_COLOR
+        color_modes = attributes.get(light.ATTR_SUPPORTED_COLOR_MODES, [])
+        return light.COLOR_MODE_COLOR_TEMP in color_modes or any(
+            mode in color_modes for mode in light.COLOR_MODES_COLOR
```

Comment:
Turning around would be faster because `light.COLOR_MODES_COLOR` is a set and `color_modes` is most likely going to be shorter.